code/modules/overmap/exoplanets/decor/_turfs.dm

```diff
@@ -34,17 +35,18 @@
 			if(exoplanet.planetary_area && istype(loc, world.area))
 				ChangeArea(src, exoplanet.planetary_area)
 		// if away site
-		else if(istype(site, /datum/map_template/ruin/away_site))
-			var/datum/map_template/ruin/away_site/away_site = site
+		else if(istype(template, /datum/map_template/ruin/away_site))
+			var/datum/map_template/ruin/away_site/away_site = template
 			if(away_site.exoplanet_atmosphere)
 				initial_gas = away_site.exoplanet_atmosphere.gas.Copy()
 				temperature = away_site.exoplanet_atmosphere.temperature
 			else
 				initial_gas = list()
 				temperature = T0C
-			set_light(MINIMUM_USEFUL_LIGHT_RANGE, away_site.lightlevel, away_site.lightcolor)
+			if(is_outside())
```

Comment:
This is a MASSIVE performance penality on such an heavy proc

Reply:
`is_outside()`, it's called on New but only if there's a planetary area and said area is space, which doesn't seem like a lot, or for mobs that are affected by weather, that also aren't that many

Here, on the other hand, would be called about 255x255 = ~65k times to initialize a single exoplanet, which is also not the only exoplanet most likely; we can't avoid setting the light, but we can probably avoid calling this proc since we weren't before, why was this check added?

Reply:
this section of code for away site based exoplanets wasn't executed at all before, cause it was broken and I only now got around to fixing it

is_outside() check is needed so only outside turfs get light (so there's no light in caves or in rooms or etc)

Reply:
changed the check to `if(away_site.exoplanet_lightlevel && is_outside())`